src/librustc_mir/build/scope.rs

```diff
@@ -138,21 +144,7 @@ struct DropData<'tcx> {
 }
 
 #[derive(Debug, Default, Clone, Copy)]
-pub(crate) struct CachedBlock {
-    /// The cached block for the cleanups-on-diverge path. This block
-    /// contains code to run the current drop and all the preceding
-    /// drops (i.e., those having lower index in Drop’s Scope drop
-    /// array)
-    unwind: Option<BasicBlock>,
-
-    /// The cached block for unwinds during cleanups-on-generator-drop path
-    ///
-    /// This is split from the standard unwind path here to prevent drop
-    /// elaboration from creating drop flags that would have to be captured
-    /// by the generator. I'm not sure how important this optimization is,
-    /// but it is here.
-    generator_drop: Option<BasicBlock>,
```

Comment:
Why is this removed?

Reply:
The effect of this commit is to undo the optimization mentioned in the comment. We need the drop flags (`StorageDead` statements) that it removed, in order to do the analysis for the generator layout optimization.

Reply:
So this unifies the `unwind` and `generator_drop` paths again.

Reply:
> The effect of this commit is to undo the optimization mentioned in the comment. We need the drop flags (`StorageDead` statements) that it removed, in order to do the analysis for the generator layout optimization.

That doesn't make sense. `StorageDead` is unrelated to drop flags. We do not want drop flags to be created.
Additionally `unwind` has to drop generator arguments (we probably want to add those back) while the `generator_drop` path does not have them. I think you should just generate `StorageDead` statements for the `generator_drop` path, but not do the other changes.

Reply:
Oh I see, yeah I completely misunderstood what was going on here.

Still, from looking at the code I couldn't find any places where the two paths are handled any differently (after my change to generate `StorageDead` statements for the `generator_drop` path, which is why I thought this change was okay. So, I guess I'm still not 100% sure what's going on here :)

Reply:
I'll split up the commit so this is easier to see.

Reply:
I split up the commit and reverted this change (but left it in the history for now).